packages/api/src/command/medical/document/process-doc-query-webhook.ts

```diff
@@ -15,37 +15,86 @@ const { log } = out(`Doc Query Webhook`);
 const isSandbox = Config.isSandbox();
 const DOWNLOAD_WEBHOOK_TYPE = "medical.document-download";
 const CONVERSION_WEBHOOK_TYPE = "medical.document-conversion";
+type WebhookType = typeof DOWNLOAD_WEBHOOK_TYPE | typeof CONVERSION_WEBHOOK_TYPE;
+
+export const handleWebhookBeingSent = async ({
+  patient,
+  requestId,
+}: {
+  patient: Patient;
+  requestId: string;
+}) => {
+  const downloadStatus = patient.data?.documentQueryProgress?.download?.status;
+  const convertStatus = patient.data?.documentQueryProgress?.convert?.status;
+
+  const isDownloadFinished = downloadStatus === "completed" || downloadStatus === "failed";
+  const isConvertFinished = convertStatus === "completed" || convertStatus === "failed";
+
+  if (isDownloadFinished) {
+    await processDocQueryProgressWebhook({
+      patient,
+      requestId,
+      progressType: "download",
+      status: downloadStatus,
+    });
+  } else if (isConvertFinished) {
+    await processDocQueryProgressWebhook({
+      patient,
+      requestId,
+      progressType: "convert",
+      status: convertStatus,
+    });
+  }
+};
 
 /**
  * Processes the document query progress to determine if when to send the document download and conversion webhooks
  */
 export const processDocQueryProgressWebhook = async ({
   patient,
-  documentQueryProgress,
   requestId,
+  progressType,
+  status,
 }: {
   patient: Pick<Patient, "id" | "cxId" | "externalId">;
-  documentQueryProgress: DocumentQueryProgress;
   requestId: string;
+  progressType: ProgressType;
+  status: DocumentQueryStatus;
 }): Promise<void> => {
   const { id: patientId } = patient;
 
   try {
     const webhooks = await getAllWebhookRequestByRequestId(requestId);
```

Comment:
Just noting: this is very expensive and we're calling it all the time.

https://github.com/metriport/metriport-internal/issues/1460

Reply:
Maybe instead of doing this when we do send a webhook i can update the external data with like downloadWebhookSent or something along those lines

Reply:
Updated this logic